### ISSUE TITLE
Correct way of starting training on custom dataset

### ISSUE THREAD

**Issue by akshaypx** [WongKinYiu/YOLO#151]
### Issue Description -

Hi, I am also facing issue, when training is initiated it gets stuck on first epoch.

The dataset structure same, i.e. dataset folder in root, which contains images, labels and test folder. Each contains images and labels folder.

I have added ```data.yaml``` in ```/yolo/config/dataset/```.

This is the data yaml file contents:
```
path: dataset 
train: train 
validation: valid 

class_num: 1 
nc: 1 
names: ["tws"] 
```

The command I used is as mentioned in the docs : ``` python yolo/lazy.py task=train dataset=data use_wandb=True ```.

Inside train yaml file, the epochs have been set to 2 just to check this issue and it takes more than one hour and crashes(as seen in wandb status) or goes on.

_Originally posted by @akshaypx in https://github.com/WongKinYiu/YOLO/issues/148#issuecomment-2571674319_

There is no progress in the progress bars as well in the terminal.

![image](https://github.com/user-attachments/assets/f2cab4c8-9377-41be-8dae-cc09e2287468)

Please correct me if I am wrong in any step.

**Comment by tahsinalamin** [comment]
Followed the same steps and it worked for me. However, added `device=cuda` to the command. Also, `use_wandb` is `True` by default. 

`python yolo/lazy.py task=train dataset=data device=cuda`

**Comment by SIDD-1991** [comment]
Could you please elaborate more on how and where you put the training data what is the directory structure and are you using txt annotations?

**Comment by SIDD-1991** [comment]
my issue is that i have started to train with custom dataset and i am unaware of how and where to put my images and label's and in what format should i put it.
What files should i modify and what commands to use.
